skins/oasis/modules/LatestActivityController.class.php

```diff
@@ -1,53 +1,51 @@
 <?php
+
 class LatestActivityController extends WikiaController {
 
 	public function init() {
 		$this->userName = '';
 	}
 
-	public function executeIndex() {
-		wfProfileIn(__METHOD__);
+	public function index() {
+		wfProfileIn( __METHOD__ );
 		$maxElements = 4;
 
-		global $wgLang, $wgContentNamespaces, $wgMemc, $wgUser;
-		$this->moduleHeader = wfMsg('oasis-activity-header');
-		
-		if( empty($this->userName) ) {
-			$mKey = wfMemcKey('mOasisLatestActivity');
-			$feedData = $wgMemc->get($mKey);
+		if ( empty( $this->userName ) ) {
+			$mKey = wfMemcKey( 'mOasisLatestActivity' );
+			$feedData = $this->wg->Memc->get( $mKey );
 		}
-		
-		if (empty($feedData)) {
+
+		if ( empty( $feedData ) ) {
 			// data provider
-			$includeNamespaces = implode('|', $wgContentNamespaces);
-			$parameters = array(
+			$includeNamespaces = implode( '|', $this->wg->ContentNamespaces );
+			$parameters = [
 				'type' => 'widget',
 				//'tagid' => $id,
 				'maxElements' => $maxElements,
-				'flags' => array('shortlist'),
-				'uselang' => $wgLang->getCode(),
-				'includeNamespaces' => $includeNamespaces
-			);
-			
-			$feedProxy = new ActivityFeedAPIProxy($includeNamespaces, $this->userName);
-			$feedProvider = new DataFeedProvider($feedProxy, 1, $parameters);
-			$feedData = $feedProvider->get($maxElements);
-			if( empty($this->userName) ) {
-				$wgMemc->set($mKey, $feedData);
+				'flags' => [ 'shortlist' ],
+				'uselang' => $this->wg->Lang->getCode(),
+				'includeNamespaces' => $includeNamespaces,
+			];
+
+			$feedProxy = new ActivityFeedAPIProxy( $includeNamespaces, $this->userName );
+			$feedProvider = new DataFeedProvider( $feedProxy, 1, $parameters );
+			$feedData = $feedProvider->get( $maxElements );
+			if ( empty( $this->userName ) ) {
+				$this->wg->Memc->set( $mKey, $feedData );
 			}
 		}
-		
+
 		// Time strings are slow to calculate, but we still want them to update frequently (60 seconds)
-		if( empty($this->userName)) {
-			$mKeyTimes = wfMemcKey('mOasisLatestActivity_times', $wgLang->getCode());
-			$this->changeList = $wgMemc->get($mKeyTimes, array());
+		if ( empty( $this->userName ) ) {
+			$mKeyTimes = wfMemcKey( 'mOasisLatestActivity_times', $this->wg->Lang->getCode() );
+			$this->changeList = $this->wg->Memc->get( $mKeyTimes ) ?? [];
 		}
-		
-		if(empty($this->changeList) && !empty($feedData) && is_array($feedData['results'])) {
-			$changeList = array();
-			foreach( $feedData['results'] as $change ) {
-				$item = array();
-				$item['time_ago'] = wfTimeFormatAgoOnlyRecent($change['timestamp']); // TODO: format the timestamp on front-end to allow longer caching in memcache?
+
+		if ( empty( $this->changeList ) && !empty( $feedData ) && is_array( $feedData['results'] ) ) {
+			$changeList = [ ];
+			foreach ( $feedData['results'] as $change ) {
+				$item = [ ];
+				$item['time_ago'] = wfTimeFormatAgoOnlyRecent( $change['timestamp'] ); // TODO: format the timestamp on front-end to allow longer caching in memcache?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)